frontend/spec/features/checkout_spec.rb

```diff
@@ -498,9 +498,18 @@
     end
   end
 
-  context "order has only payment step" do
+  context "order has only payment step and does not build shipments" do
+    class DisallowShipments
+      def self.call(_order)
+        false
+      end
+    end
+
     before do
       create(:credit_card_payment_method)
+      @original_predicate_class = Spree::Config.build_shipment_predicate_class
```

Comment:
why not using `let!` here?

Reply:
I thought it would be simpler to leave it in the same `before` and `after` blocks where the rest of the setup and teardown were taking place. I can move it out to a `let!` if that's preferred.

Reply:
I tend to avoid instance variables in specs as much as possible and I'd like to even discourage future developers to use them (watching them here could make them feel like it's ok to use). I understand here it makes sense though. Maybe using `around` instead of `before/after` would fix the issue and will avoid using any instance variable?